src/kundera-mongo/src/main/java/com/impetus/client/mongodb/MongoDBClient.java

```diff
@@ -1457,7 +1457,11 @@ else if (jsonClause.contains("mapReduce("))
     }
 
     private MapReduceCommand parseMapReduceCommand(String jsonClause) {
-        String collectionName = jsonClause.replaceFirst("(?ms).*\\.\\s*([^.]+?)\\s*\\.\\s*mapReduce\\s*\\(.*", "$1");
+        String collectionName = jsonClause.replaceFirst("(?ms).*?\\.\\s*(.+?)\\s*\\.\\s*mapReduce\\s*\\(.*", "$1");
+        if (collectionName.contains("getCollection")) {
+            collectionName = collectionName.replaceFirst(".*getCollection\\s*\\(\\s*['\"](.*)['\"]\\s*\\).*", "$1");
```

Comment:
Good point!
Though using a different quote character would make the expression syntactically incorrect.
I'll change it anyway and resubmit.
Thanks!